src/async_impl/client.rs

```diff
@@ -1098,6 +1098,31 @@ impl Client {
     }
 }
 
+#[cfg(feature = "cookies")]
+impl Client {
+    /// Returns a reader lock on the `CookieStore` (immutable access).
+    /// Requests will be blocked until the lock is dropped.
+    pub fn get_cookies(
+        &self
+    ) -> Option<std::sync::RwLockReadGuard<'_, cookie::CookieStore>> {
+        match self.inner.cookie_store.as_ref()?.read() {
+            Ok(cookies) => Some(cookies),
+            Err(e) => Some(e.into_inner()),
+        }
+    }
+
+    /// Returns a writer lock on the `CookieStore` (mutable access).
+    /// Requests will be blocked until the lock is dropped.
+    pub fn get_cookies_mut(
```

Comment:
I think mutating the client cookie store could be confusing, since it's a shared resource internally to all client clones. What if we just start with being able to export the cookie store, and eventually being able to build a new `Client` with a store in the builder?

Reply:
Yes, a CookieStore in the builder is a great idea. But it could be useful to mutate the cookie store. For example, some websites have a client-side script settings cookies. It would be great to be able to simulate this.